include/grpcpp/security/credentials.h

```diff
@@ -110,10 +110,7 @@ class ChannelCredentials : private grpc::internal::GrpcLibrary {
     return nullptr;
   }
 
-  // TODO(yashkt): This is a hack that is needed since InsecureCredentials can
-  // not use grpc_channel_credentials internally and should be removed after
-  // insecure builds are removed from gRPC.
-  virtual bool IsInsecure() const { return false; }
```

Comment:
I think we should also be able to remove the `AsSecureCredentials()` method.

Reply:
I had considered it, and stopped short of doing the `AsSecureCredentials` removal because of the implications on `CompositeChannelCredentials` and `CompositeCallCredentials`. I agree the AsSecureCredentials API doesn't seem quite right, but without that, I think we'll need to do some c_creds check against other security definitions, which just kicks the can down the road. I'm also not sure what the scope of AsSecureCredentials is intended to be in this context ... is it anything that's not explicitly SecureCredentials, or eventually broader? Currently, only SecureCredentials return a value. An explicit check against InsecureCredentials would maybe not be safe though. (cronet & binder return nullptr today, but if someday they don't, this logic would have to change)

I suggest leaving this as a separate work. This PR can be merged without it.